forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -1162,6 +1162,11 @@ else if (defined.equals("EnchantedPlayer")) {
             if (o instanceof Player) {
                 players.add((Player) o);
             }
+        } else if (defined.equals("EnchantedController")) {
```

Comment:
shouldn't EnchantedController be handled by the part below Line 1166/1171 ? the addPlayer part should handle Controller

Reply:
hmm, sadly it doesn't
```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.base/java.util.LinkedList.checkElementIndex(LinkedList.java:559)
	at java.base/java.util.LinkedList.get(LinkedList.java:480)
	at forge.util.collect.FCollection.get(FCollection.java:396)
	at forge.game.staticability.StaticAbilityMustAttack.entitiesMustAttack(StaticAbilityMustAttack.java:28)
	at forge.game.combat.AttackRequirement.<init>(AttackRequirement.java:54)
	at forge.game.combat.AttackConstraints.<init>(AttackConstraints.java:95)
	at forge.game.combat.Combat.initConstraints(Combat.java:135)
	at forge.game.combat.Combat.<init>(Combat.java:89)
	at forge.game.phase.PhaseHandler.onPhaseBegin(PhaseHandler.java:314)
	at forge.game.phase.PhaseHandler.startFirstTurn(PhaseHandler.java:1125)
	at forge.game.GameAction.startGame(GameAction.java:1992)
	at forge.game.Match.startGame(Match.java:90)
	at forge.gamemodes.match.HostedMatch$2.run(HostedMatch.java:253)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

Reply:
you need to pass the stAb host not the attacker into getDefined I think